### PR TITLE
zib-AdvanceDirective

### PR DESCRIPTION
**Pull request by ArdonToonstra**
Hola! Zouden jullie deze willen reviewen? 
Ik heb nog één error, maar dat is geloof ik een fout in de FHIR spec? https://chat.fhir.org/#narrow/stream/179166-implementers/topic/Consent.2Escope.3A.20diff.20in.20codes.20shown.20on.20FHIR.20vs.20Tx.20server

### CODE REVIEW

File: resources/zib/zib-AdvanceDirective.xml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  <kind value="resource" />
+  <abstract value="true" />
+  <type value="Consent" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Consent" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Consent">
+      <path value="Consent" />
+      <short value="AdvanceDirective" />
+      <definition value="Root concept of the LivingWill information model. This concept contains all data elements of the LivingWill information model." />
```
Comment (LuudSlagter) [inline resources/zib/zib-AdvanceDirective.xml:35]: Based on https://informatiestandaarden.nictiz.nl/wiki/FHIR:V1.0_FHIR_Profiling_Guidelines_R4#StructureDefinitions (11.2.1.2, second bullet), this definition should not be taken from the zib, as it does not provide meaningful additional information.

File: release-notes.md [hunk 2]
```diff
@@ -13,6 +13,15 @@ This document contains release notes per zib, indicating differences with their
  * The notion to mark an address as 'official' seems to have its origin in the BRP, but is absent in the zib model.
  * The addition of `extension:official` seemed based on the rationale: "if implementers do not support the custom `extension:addressType`, you should be able to determine the AddressType by looking at `.use`, `.type` and custom `extension:official`". However, in this use case extension:official would probably also be not supported. By mapping 'HP' to .type 'both', all AddressType codes can be mapped to a combination of .type and .use.
 
+## zib-AdvanceDirective
+* Moved TypeOfLivingWill from `Consent.category` to `Consent.provision.code` and made the element 0..1 rather than 1..1 to adhere to the conceptual cardinalities of the zib.
+* Moved LivingWillDocument from `Consent.source[x]` to `Consent.source[x]:sourceAttachment.data`.
+* Moved Representative from `Consent.consentingParty` to `Consent.provision.actor:representative.reference` because the element is removed in the R4. Also `Consent.provision.actor:representative.role` has now a fixed value for this mapping.
```
Comment (LuudSlagter) [inline release-notes.md:19]: Replace 'the R4' by 'R4' (sorry for the major nitpicking)

File: resources/zib/zib-AdvanceDirective.xml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      <alias value="WilsverklaringDocument" />
+      <mapping>
+        <identity value="zib-advancedirective-v3.1.1-2020EN" />
+        <map value="NL-CM:7.15.5" />
+        <comment value="LivingWillDocument" />
+      </mapping>
+    </element>
+    <element id="Consent.policy.uri">
+      <path value="Consent.policy.uri" />
+      <comment value="Either a policy or policyRule is mandated by the HL7 core resourece by the ppc-1 constraint, therefore, if no explicit value is recorded the fallback is to populate `policy.uri` with 'http://wetten.overheid.nl/'." />
```
Comment (LuudSlagter) [inline resources/zib/zib-AdvanceDirective.xml:152]: Replace 'resourece' by 'resource'.

File: release-notes.md [hunk 4]
```diff
@@ -13,6 +13,15 @@ This document contains release notes per zib, indicating differences with their
  * The notion to mark an address as 'official' seems to have its origin in the BRP, but is absent in the zib model.
  * The addition of `extension:official` seemed based on the rationale: "if implementers do not support the custom `extension:addressType`, you should be able to determine the AddressType by looking at `.use`, `.type` and custom `extension:official`". However, in this use case extension:official would probably also be not supported. By mapping 'HP' to .type 'both', all AddressType codes can be mapped to a combination of .type and .use.
 
+## zib-AdvanceDirective
+* Moved TypeOfLivingWill from `Consent.category` to `Consent.provision.code` and made the element 0..1 rather than 1..1 to adhere to the conceptual cardinalities of the zib.
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply release-notes.md:17]: Scherp!

File: resources/zib/zib-AdvanceDirective.xml [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+      <alias value="WilsverklaringDocument" />
+      <mapping>
+        <identity value="zib-advancedirective-v3.1.1-2020EN" />
+        <map value="NL-CM:7.15.5" />
+        <comment value="LivingWillDocument" />
+      </mapping>
+    </element>
+    <element id="Consent.policy.uri">
+      <path value="Consent.policy.uri" />
+      <comment value="Either a policy or policyRule is mandated by the HL7 core resource by the ppc-1 constraint, therefore, if no explicit value is recorded a fallback is to populate `policy.uri` with _http://wetten.overheid.nl/_." />
```
Comment (jduwel) [inline resources/zib/zib-AdvanceDirective.xml:151]: Ik zou het iets anders verwoorden, met toch ook de belangrijke nuance dat de suggestie alleen van toepassing is als ook daadwerkelijk de Nederlandse wet wordt gevolgd.

> Either `.policy` or `.policyRule` should exist following core constraint ppc-1. If no explicit value is recorded, a fallback is to populate `policy.uri` with 'https://wetten.overheid.nl/' if the advance directive is recorded following Dutch law and regulations.

File: resources/zib/ext-AdvanceDirective.Disorder.xml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      <fixedUri value="http://nictiz.nl/fhir/StructureDefinition/ext-AdvanceDirective.Disorder" />
+    </element>
+    <element id="Extension.value[x]">
+      <path value="Extension.value[x]" />
+      <short value="Disorder" />
+      <definition value="The problem or disorder to which the living will applies." />
+      <alias value="Aandoening" />
+      <type>
+        <code value="Reference" />
+        <targetProfile value="http://nictiz.nl/fhir/StructureDefinition/zib-Problem" />
```
Comment (jduwel) [inline resources/zib/ext-AdvanceDirective.Disorder.xml:47]: Should this be nl-core or not? See remark in #169

File: resources/zib/zib-AdvanceDirective.xml [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  <publisher value="Nictiz" />
+  <contact>
+    <name value="Nictiz" />
+    <telecom>
+      <system value="email" />
+      <value value="info@nictiz.nl" />
+      <use value="work" />
+    </telecom>
+  </contact>
+  <description value="A living will is a verbal or written description of the patient’s wishes with regard to future medical action or end of their life. A living will is mainly used for situations in which the patient is no longer able to speak about these decisions with their healthcare provider." />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-AdvanceDirective.xml:17]: https://bits.nictiz.nl/browse/ZIB-1597

File: resources/zib/zib-AdvanceDirective.xml [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+        <map value="NL-CM:7.15.7" />
+        <comment value="LivingWillDate (implicit, main mapping is on `Consent.dateTime`)" />
+      </mapping>
+    </element>
+    <element id="Consent.scope">
+      <path value="Consent.scope" />
+      <patternCodeableConcept>
+        <coding>
+          <system value="http://terminology.hl7.org/CodeSystem/consentscope" />
+          <code value="adr" />
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply resources/zib/zib-AdvanceDirective.xml:97]: Zie opmerking bij change log.

File: resources/zib/zib-AdvanceDirective.xml (hunk 5 above)
Comment (ArdonToonstra) [reply resources/zib/zib-AdvanceDirective.xml:151]: Nice! Aangepast.

File: resources/zib/ext-AdvanceDirective.Disorder.xml (hunk 6 above)
Comment (ArdonToonstra) [reply resources/zib/ext-AdvanceDirective.Disorder.xml:47]: Yes, this should be core!Thnx.

File: resources/zib/zib-AdvanceDirective.xml [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+      <alias value="Toelichting" />
+      <mapping>
+        <identity value="zib-advancedirective-v3.1.1-2020EN" />
+        <map value="NL-CM:7.15.6" />
+        <comment value="Comment" />
+      </mapping>
+    </element>
+    <element id="Consent.status">
+      <path value="Consent.status" />
+      <comment value="This element is implictly mapped to the zib concept NL-CM:7.15.7 (LivingWillDate). Unless the status is explicitly recorded, the following guidance applies:&#xD;&#xA;&#xD;&#xA;* When the LivingWillDate is in the future, `.status` will usually be set to _proposed_.&#xD;&#xA;* When LivingWillDate is in the past, `.status` will usually be set to _active_.&#xD;&#xA;* When LivingWillDate is absent, which is not possible by the zib, `.status` will usually be set to _draft_." />
```
[the remark this answers is not in the document]
Comment (jduwel) [reply resources/zib/zib-AdvanceDirective.xml]: Ja dat laatste, "tenzij het systeem het beter weet"

File: known-issues.yml [hunk 10]
```diff
@@ -28,6 +34,16 @@ zib-AddressInformation:
       - datatype: string
         reason: The mapping of zib AddressInformation on the FHIR Address datatype is the result of compatibility with HL7v3, which is the format that a lot of healthcare data in the Netherlands is stored in.  As a result of this, the zib HouseNumberIndication concept with CD datatype is mapped to a FHIR string datatype with a constraint added (the value can only be 'by' or 'to').
 
+zib-AdvanceDirective:
+  ignored issues:
+    StructureDefinition.differential.element[6].pattern.ofType(CodeableConcept).coding[0]:
```
[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply known-issues.yml]: Didn't know you could use FHIRpath notation here too, cool! Will do next time.
This one is deleted so I will resolve this conversation.

**Comment by ArdonToonstra** [comment]
Nog een link toegevoegd voor de nu bekende validatie error, kan deze dan al wel naar main?

File: known-issues.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (pieter-edelman-nictiz) [reply known-issues.yml]: Whelp! No you can't use FHIRpath, but you can use the path as found on the id of the element.

[the remark this answers is not in the document]
Comment (ArdonToonstra) [reply known-issues.yml]: ah, haha, I read it wrong I guess. Thnx for the clarification.

**Comment by pieter-edelman-nictiz** [comment]
The missing "adr" in http://terminology.hl7.org/CodeSystem/consentscope keeps generating validation problems, not only in the profiles but also in the examples. I think we should think a bit harder how to deal with this before merging this to main.
At the moment I see two options:
* Add the corrected CodeSystem to our package. This would mean we "publish" something that we don't own but we also know that this is a very special case where the official source is broken.
* Wait for #178 to be solved to properly handle these kind of errors.

**Comment by pieter-edelman-nictiz** [comment]
Manually merged

**Pull request closed without merging**